ui/src/app/services/sign-in.service.ts

```diff
@@ -0,0 +1,113 @@
+import 'rxjs/Rx';
+import {Injectable, NgZone} from '@angular/core';
+import {Observable} from 'rxjs/Observable';
+import {BehaviorSubject} from 'rxjs/BehaviorSubject';
+import {User} from 'app/models/user';
+
+declare const gapi: any;
+
+interface BasicProfile {
+  id: string;
+  name: string;
+  givenName: string;
+  familyName: string;
+  imageUrl: string;
+  email: string;
+}
+
+export interface SignInDetails {
```

Comment:
Do we need all these fields on the client?

I'm working on a "me" endpoint right now... should we return all of these from it?